python/plugins/processing/core/outputs.py

```diff
@@ -250,15 +250,34 @@ class OutputVector(Output):
     encoding = None
     compatible = None
 
-    def getFileFilter(self, alg):
+    def __init__(self, name='', description='', hidden=False, base_input=None):
+        Output.__init__(self, name, description, hidden)
+        self.base_input = base_input
+        self.base_layer = None
+
+    def hasGeometry(self):
+        if self.base_layer is None:
+            return True
+        return dataobjects.canUseVectorLayer(self.base_layer, [-1])
+
+    def getSupportedExtensions(self):
```

Comment:
I'm with Victor here, better to be consistent with other classes/packages. We have `getSupportedOutputRasterLayerExtensions`, `getSupportedOutputVectorLayerExtensions` and `getSupportedOutputTableExtensions` in other places